service/matching/backlog_manager_test.go

```diff
@@ -154,3 +163,146 @@ func TestTaskWriterShutdown(t *testing.T) {
 	err = tlm.backlogMgr.SpoolTask(&persistencespb.TaskInfo{})
 	require.Error(t, err)
 }
+
+func TestApproximateBacklogCountIncrement_taskWriterLoop(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	backlogMgr := newBacklogMgr(controller)
+
+	// Adding tasks on the taskWriters channel
+	backlogMgr.taskWriter.appendCh <- &writeTaskRequest{
+		taskInfo: &persistencespb.TaskInfo{
+			ExpiryTime: timestamp.TimeNowPtrUtcAddSeconds(3000),
+			CreateTime: timestamp.TimeNowPtrUtc(),
+		},
+		responseCh: make(chan<- *writeTaskResponse),
+	}
+
+	require.Equal(t, backlogMgr.db.getApproximateBacklogCount(), int64(0))
+
+	backlogMgr.taskWriter.Start()
+	// Adding tasks to the buffer shall increase the in-memory counter by 1
```

Comment:
here, "adding" actually might be correct, because you're explaining the effects of the tasks being added, which is currently happening as you're waiting in the `Eventually`. I hope that makes sense